constructr-coordination-consul/src/main/scala/com.tecsisa.constructr.coordination.consul/ConsulCoordination.scala

```diff
@@ -81,53 +101,59 @@ final class ConsulCoordination(
     Http(system).outgoingConnection(host, port)
   }
 
-  override def getNodes[A: NodeSerialization]() = {
-    def unmarshalNodes(entity: ResponseEntity) = {
+  override def getNodes() = {
+    def unmarshalNodes(entity: ResponseEntity): Future[Set[Address]] = {
       def toNodes(s: String) = {
         import rapture.json._
         import rapture.json.jsonBackends.circe._
         def jsonToNode(json: Json) = {
           val init = nodesUri.path.toString.stripPrefix(kvUri.path.toString)
           val key = json.Key.as[String].substring(init.length)
-          implicitly[NodeSerialization[A]].fromBytes(Base64.getUrlDecoder.decode(key))
+          val uri = new String(getUrlDecoder.decode(key), UTF_8)
+          AddressFromURIString(uri)
         }
         Json.parse(s).as[Set[Json]].map(jsonToNode)
       }
       Unmarshal(entity).to[String].map(toNodes)
     }
     val uri = nodesUri.withQuery(Uri.Query("recurse"))
     send(Get(uri)).flatMap {
-      case HttpResponse(OK, _, entity, _)       => unmarshalNodes(entity)
-      case HttpResponse(NotFound, _, entity, _) => ignore(entity).map(_ => Set.empty)
-      case HttpResponse(other, _, entity, _)    => ignore(entity).map(_ => throw UnexpectedStatusCode(uri, other))
+      case HttpResponse(OK, _, entity, _) => unmarshalNodes(entity)
+      case HttpResponse(NotFound, _, entity, _) =>
+        ignore(entity).map(_ => Set.empty)
+      case HttpResponse(other, _, entity, _) =>
+        ignore(entity).map(_ => throw UnexpectedStatusCode(uri, other))
     }
   }
 
-  override def lock[A: NodeSerialization](self: A, ttl: FiniteDuration) = {
+  override def lock(self: Address, ttl: FiniteDuration) = {
     val uriLock = baseUri.withPath(baseUri.path / "lock")
     def readLock() = {
       def unmarshallLockHolder(entity: ResponseEntity) = {
         def toLockHolder(s: String) = {
           import rapture.json._
           import rapture.json.jsonBackends.circe._
-          def jsonToNode(json: Json) = {
-            val value = json.Value.as[String]
-            new String(Base64.getUrlDecoder.decode(value), UTF_8)
-          }
-          Json.parse(s).as[Set[Json]].map(jsonToNode).head
+          Json.parse(s).node.value.as[String]
```

Comment:
Why do you think that this change is needed? 
With this change, you can't lock a session properly. 
This is because consul API returns a json list and not a single object. 
Also we need to decode the value using base64 because we received an encoded value.

`http http://localhost:8500/v1/kv/constructr/ConstructR-Consul/lock`

```
HTTP/1.1 200 OK
Content-Length: 221
Content-Type: application/json
Date: Thu, 10 Nov 2016 08:39:39 GMT
X-Consul-Index: 9
X-Consul-Knownleader: true
X-Consul-Lastcontact: 0

[
    {
        "CreateIndex": 9, 
        "Flags": 0, 
        "Key": "constructr/ConstructR-Consul/lock", 
        "LockIndex": 1, 
        "ModifyIndex": 9, 
        "Session": "3230a44c-0b7d-0e23-ba6b-35554118ee94", 
        "Value": "YWtrYS50Y3A6Ly9Db25zdHJ1Y3RSLUNvbnN1bEAxNzIuMTkuMC41OjI1NTI="
    }
]
```